pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -258,3 +258,215 @@ type TiKVFailureStore struct {
+}
+
+// cephStorageProvider represents an ceph compatible bucket for storing backups.
+type CephStorageProvider struct {
+	// Region in which the ceph bucket is located.
+	Region string `json:"region"`
+	// Bucket in which to store the Backup.
+	Bucket string `json:"bucket"`
+	// Endpoint is the access address of the ceph object storage.
+	Endpoint string `json:"endpoint"`
+	// SecretName is the name of secret which stores
+	// ceph object store access key and secret key.
+	SecretName string `json:"secretName"`
+}
+
+// BackupType respresents the backup type.
+type BackupType string
+
+const (
+	// BackupTypeFull respresents the full backup of tidb cluster.
+	BackupTypeFull BackupType = "full"
+	// BackupTypeInc respresents the incremental backup of tidb cluster.
+	BackupTypeInc BackupType = "incremental"
+)
+
+// BackupSpec contains the backup specification for a tidb cluster.
+type BackupSpec struct {
+	// StorageType is the backup storage type.
+	StorageType BackupStorageType `json:"storageType"`
+	// StorageProvider configures where and how backups should be stored.
+	StorageProvider `json:",inline"`
+	// Type is the backup type for tidb cluster.
+	Type BackupType `json:"backupType"`
+	// Cluster is the Cluster to backup.
+	Cluster string `json:"cluster"`
+	// SecretName is the name of secret which stores
+	// tidb cluster's username and password.
+	SecretName string `json:"secretName"`
+}
+
+// BackupConditionType represents a valid condition of a Backup.
+type BackupConditionType string
+
+const (
+	// BackupRunning means the Backup is currently being executed.
+	BackupRunning BackupConditionType = "Running"
+	// BackupComplete means the Backup has successfully executed and the
+	// resulting artifact has been stored in backend storage.
+	BackupComplete BackupConditionType = "Complete"
+	// BackupFailed means the Backup has failed.
+	BackupFailed BackupConditionType = "Failed"
+)
+
+// BackupCondition describes the observed state of a Backup at a certain point.
+type BackupCondition struct {
+	Type    BackupConditionType
+	Status  corev1.ConditionStatus
+	Reason  string
+	Message string
+}
+
+// BackupStatus represents the current status of a backup.
+type BackupStatus struct {
+	// BackupPath is the location of the backup.
+	BackupPath string `json:"backupPath"`
+	// TimeStarted is the time at which the backup was started.
+	TimeStarted metav1.Time `json:"timeStarted"`
+	// TimeCompleted is the time at which the backup completed.
+	TimeCompleted metav1.Time `json:"timeCompleted"`
+	// BackupSize is the data size of the backup.
+	BackupSize int64 `json:"backupSize"`
+	// CommitTs is the snapshot time point of tidb cluster.
+	CommitTs   string            `json:"backupSize"`
+	Conditions []BackupCondition `json:"conditions"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupSchedule is a backup schedule of tidb cluster.
+type BackupSchedule struct {
```

Comment:
This should be `ScheduledBackup`

Reply:
I think called `BackupSchedule` is ok, the periodic backup of other projects is also named